Send welcome email after verification email

dvorakjt opened the pull request:
## Overview 
This PR moves the send welcome email feature into the verification success component, correcting the issue of the welcome email arriving prior to or at the same time as the verification email.

## Test Plan 
This PR was tested manually in the browser by going through the sign up and verify flow.

## Follow ups 
This branch has conflicts which must be resolved. We will just need to verify it is working as expected in production.

mnarayan1 committed:
created welcome and actions page for player

mnarayan1 committed:
added voter registration page

mnarayan1 committed:
player goes to rock the vote of login

mnarayan1 committed:
player goes to rock the vote or login

mnarayan1 committed:
Merge branch 'development' of https://github.com/mnarayan1/frontend into development

mnarayan1 committed:
added basic styling to player pages

mnarayan1 committed:
fixed routing

Devanshusp committed:
fixed component error

Devanshusp committed:
Merge branch '8by8-org:development' into development

Devanshusp committed:
made header fixed and adjusted navbar animation

Devanshusp committed:
merge conflict

Devanshusp committed:
fixed header spacing error

Devanshusp committed:
added scss styles folder

Devanshusp committed:
added /challengerwelcome page

mnarayan1 committed:
Merge branch 'development' into development

Devanshusp committed:
moved styles folder out of components folder

Devanshusp committed:
added files in styles folder

Devanshusp committed:
Merge pull request #35 from Devanshusp/development

[0.1 & 1.1] Created style folder and added /challengerwelcome page

mnarayan1 committed:
added election reminders and challenge for player

kphan20 committed:
Fixed merge conflicts

kphan20 committed:
Merge branch '8by8-org-development' into development after merge
conflict resolution

kphan20 committed:
Added PrivateRoute to progress page

kphan20 committed:
Added avatar image specifications and resolves sass merge conflict

kphan20 committed:
Added another test badge

Devanshusp committed:
cleaned up unused challenge page and renamed avatars

Devanshusp committed:
Merge pull request #34 from kphan20/development

Task 1.5: Build the /progress page

dvorakjt committed:
integrate RTV Election Reminder form (#46)

Devanshusp committed:
removed previously deleted component not found

Devanshusp committed:
changed assets folder stucture

Devanshusp committed:
added font and button styles

Devanshusp committed:
updated /homepage design

Devanshusp committed:
fixed svg spacing

Devanshusp committed:
added footer with social media links

Devanshusp committed:
cursor pointer on footer icon hover

mnarayan1 committed:
player redirected to action after login

Devanshusp committed:
Updated styling on /homepage + added footer (#48)

* changed assets folder stucture

* added font and button styles

* updated /homepage design

* fixed svg spacing

* added footer with social media links

* cursor pointer on footer icon hover

mnarayan1 committed:
fixed issues

Devanshusp committed:
uncommented homepage scss file

Devanshusp committed:
typo

Devanshusp committed:
fixed merge conflicts

Devanshusp committed:
Merge branch 'mnarayan1-development' into development

Devanshusp committed:
Merge branch 'development' into development

Devanshusp committed:
Merge pull request #1 from 8by8-org/development

Updated styling on /homepage + added footer (#48)

arniber21 committed:
Task 1.6, Task 4.2, Task 0.2 (#32)

* Task 1.6: Build the /signin page and workflow (switching back and forth to signup)

Basic login system works. Login/Logout/Signup loop exists. No email verification yet (this probably needs to happen sometime).

* Task 2.1

build a link generator that makes a unique link based on a signed-in user profile (also implement the reverse mechanism: from a url to a user’s identifier, e.g. uid, email)

* Cleaning up some code

* Fixing some firebase auth errors

* Cleaning up some imports

* Make the query param a base64 string

* Added a comment

* Changed login/signup so no password is required

* Seperated the task 2.1 functions to their own file

* Remove /playerwelcome from routes

* Added some documentation, fixed some imports

* Changes for merging

* Fixing a temp error

* Some more dependencies

* Deleting unused components

* Fixing dependencies

* Attempt to fix conflicts

* Fixing package.json

* Updated some functionality to also solve task 4.2

* fix some conflicts

* Folder Reorganization

* Attempt to fix /signup route

* Change Authentication to fit Figma

* Clean up some styles

* Change forward url to /progress

* Remove URLEncode and its test file from the PRgit

* Fix conflicts

* Fix conflicts

* Fix conflicts

* Attempt to fix merging changes

* Fix privateroute routing signout to wrong area

* Signup redirects to /progress

* Fix react warnings

* Attempt to fix conflicts

* Fixing a few small errors

* Direct which lines to delete

* fix imports

* small typo

* temp commit

* Revert "Fix react warnings"

This reverts commit 8d5d80177a2e0e1ef22695a8b282fca86dd4e14e.

* Final fixes

* Fixing typos

* Delete unused lines

* Fix small package.json error

* Some styling for Issue 1.9

* Fix some introduced conflict errors

* Revert "Fix some introduced conflict errors"

This reverts commit a03c8c216b6390039b65a511053620282304d171.

* Finish avatar selection screen

* finalize sizing

* Try to fix conflicts

* testing things out

* Fix some app errors

* Deleting unnecessary lines

* Removed unsed files

Co-authored-by: Ricky Xu <xuchen727@hotmail.com>

rickyyx committed:
Create pr_template.md

rickyyx committed:
Rename pr_template.md to pull_request_template.md

mnarayan1 committed:
styling player flow

mnarayan1 committed:
sign up links added

mnarayan1 committed:
fixes

mnarayan1 committed:
fixed action message and reminder path

Devanshusp committed:
added sharing functionality

Devanshusp committed:
Merge branch 'development' of https://github.com/Devanshusp/frontend into development

Devanshusp committed:
added styling and images

mnarayan1 committed:
1.2 & 3.1: redirection and styling for player flow (#60)

* styling player flow

* fixes

* fixed action message and reminder path

mnarayan1 committed:
Merge branch '8by8-org:development' into development

rickyyx committed:
Use Button again for button components (#65)

* Use Button again for button components

* added bootstrap buttons to all pages

Co-authored-by: mnarayan1 <madhumitas.narayan@gmail.com>

mnarayan1 committed:
Merge branch '8by8-org:development' into development

mnarayan1 committed:
generate url

Devanshusp committed:
merge confilcts

Devanshusp committed:
Merge pull request #3 from Devanshusp/8by8-org-development

merge error fix

mnarayan1 committed:
get challenger info from url

Devanshusp committed:
Task 2.3: Link Sharing (#64)

* added sharing functionality

* added styling and images

dvorakjt committed:
Task 4.3 (#63)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

mnarayan1 committed:
link generation, redirection

Devanshusp committed:
Merge branch '8by8-org:development' into development

Devanshusp committed:
added google recaptcha to signup

Devanshusp committed:
added recaptcha to signin

Devanshusp committed:
autoselect first avatar on signup

mnarayan1 committed:
link sharing with generated url

arniber21 committed:
Small Bug Fixes (#68)

* Small Bug Fixes

* Fix Signup page bugs

* Fix Signup page bugs

* Added basic Unit Tests for SignupPage

* Add basic unit tests for signup page

* Change class to className throughout project

* Fix testing bug w/ hooks

* Organize tests, fix React errors

* Fix a typo

* Add a terms of service page with basic styling

Devanshusp committed:
fixed conflicts from merge

Devanshusp committed:
Added Google ReCAPTCHA + Fixed avatar autoselect bug (#69)


* added google recaptcha to signup

* added recaptcha to signin

* autoselect first avatar on signup

Devanshusp committed:
styling for challenger welcome + bug fixes

Devanshusp committed:
Merge branch '8by8-org:development' into development

mnarayan1 committed:
Merge branch 'development' into development

mnarayan1 committed:
Revert "3.2: create sharing link and extract challenger info from link" (#82)

Devanshusp committed:
Merge branch '8by8-org:development' into development

Devanshusp committed:
firestore saves names and avatars on signup

Devanshusp committed:
properly adding name + avatar on signup

Devanshusp committed:
added firebase cloud function (needs testing) + signup styling

dvorakjt committed:
Voter Registration Feature UI (#81)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

Devanshusp committed:
added styling files + progress page

Devanshusp committed:
progress page functionality & styling

Devanshusp committed:
merge conflicts

rickyyx committed:
Update README.md

Devanshusp committed:
added env file to hide api keys

Devanshusp committed:
Added Badges in /Progress page + Emailing + Styling & Bugs (#92)

* added sharing functionality

* added styling and images

* added google recaptcha to signup

* added recaptcha to signin

* autoselect first avatar on signup

* styling for challenger welcome + bug fixes

* firestore saves names and avatars on signup

* properly adding name + avatar on signup

* added firebase cloud function (needs testing) + signup styling

* added styling files + progress page

* progress page functionality & styling

* added env file to hide api keys

wcouturiaux committed:
Update firebase install instructions (#100)

mnarayan1 committed:
3.2 link sharing update (#96)

* clean up styling

* store challenger id

* push challenger id to firebase

* push challenger to firebase working

Devanshusp committed:
added api keys doc to readme

MazharulIslam-Naim committed:
Update Header #87

MazharulIslam-Naim committed:
Update Header #87 (#106)

mnarayan1 committed:
Link update, minor styling fixes (#107)

* clean up styling

* store challenger id

* push challenger id to firebase

* push challenger to firebase working

* link is /share, styling fixes

ty730 committed:
Issue #70, IG and SMS Invite/Share workflow (#105)

* IG and SMS Invite/Share workflow, Issue #70

Created IG, SMS flow for Invite/Share, Issue #70

* Made link on one line, overflow hidden, Issue #70

Devanshusp committed:
Min width to 375px

Devanshusp committed:
challenger finishes/fails challenge

Devanshusp committed:
confetti animation on /actions

Devanshusp committed:
header icon margin

Devanshusp committed:
user type (unfinished)

Devanshusp committed:
merge conflicts

Devanshusp committed:
styling

ty730 committed:
Fixed styling turning header text black (#111)

* IG and SMS Invite/Share workflow, Issue #70

Created IG, SMS flow for Invite/Share, Issue #70

* Made link on one line, overflow hidden, Issue #70

* Fixed header text turning black

Devanshusp committed:
button change in /progress

Devanshusp committed:
fixed merge conflicts

rickyyx committed:
[icon] Fix some icon and preview messages for the website (#110)

Devanshusp committed:
merge conflicts

Devanshusp committed:
Merge pull request #3 from Devanshusp/8by8-org-development

merge conflict fixes

Devanshusp committed:
Merge branch '8by8-org:development' into development

Devanshusp committed:
NaN Days Left issue fix + getUserType function

Devanshusp committed:
Merge branch 'Devanshusp-development' into development

dvorakjt committed:
Issues 89 (complete) and 97 (partially complete) - Styling and completing Register to Vote flow, adding completedAction function (#101)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

wcouturiaux committed:
Issue #97 Partial Completion (#104)

* Update firebase install instructions

* Fix typo

* Implement completedAction for sharing challenge

* Remove unnecessary import

* Remove delay on reload

* Correct typos

* Fix invalid DOM property 'class'

MazharulIslam-Naim committed:
Merge branch 'development' into development

mnarayan1 committed:
#102 Add better loading prompt on progress (#114)

* clean up styling

* store challenger id

* push challenger id to firebase

* push challenger to firebase working

* link is /share, styling fixes

* loading wheel

* updated loading

* fix

* minor fixes, updated voter registration

* fixed signup avatars

wcouturiaux committed:
Make avatar state controlled (#118)

dvorakjt committed:
Updates to Frontend Field Validation - Register to Vote Flow (#120)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

* fixed a few bugs in the voter registration flow

* fixed merge conflict in package.json

* uncommented code that checks if the user has already registered to vote

wcouturiaux committed:
Add completedAction for challenger sharing on IG (#121)

Devanshusp committed:
Update pull_request_template.md

Devanshusp committed:
added emailing for players and challengers (#119)

* added emailing for players and challengers

* updated emailing to include challengers

MazharulIslam-Naim committed:
Update /actions page #75 (Partial) (#123)

* Added user flow through actions. Left to do: share and the confetti on the user finished all actions page.

* Fixed navbar and sidebar scaling with new min-width of root app.

* Added comments to actions page and moved it to its own folder.

* Added share component using the invite component. If the there is no challenger info in local storage, then the user gets sent to the signin page.

* Everything was just tabed over one to aline everyting properly.

* Fixed bug where there was expected data when there was no user signed in.

mnarayan1 committed:
Player can become a challenger (#129)

* clean up styling

* store challenger id

* push challenger id to firebase

* push challenger to firebase working

* link is /share, styling fixes

* loading wheel

* updated loading

* fix

* minor fixes, updated voter registration

* fixed signup avatars

* player can become challenger

CaptainExtremis committed:
Add files via upload (#132)

[P0] Added "Rock the Vote" sponsorship from Figma document.

mnarayan1 committed:
Change default select for party and race, unauthenticated users redirected to /signup (P0 #21 and #29) (#133)

* clean up styling

* store challenger id

* push challenger id to firebase

* push challenger to firebase working

* link is /share, styling fixes

* loading wheel

* updated loading

* fix

* minor fixes, updated voter registration

* fixed signup avatars

* player can become challenger

* unauthenticated users redirected to sign up

* removed default race and political party

dvorakjt committed:
Register to Vote Issues (#134)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

* fixed a few bugs in the voter registration flow

* fixed merge conflict in package.json

* uncommented code that checks if the user has already registered to vote

* separate voter registration flow into individual pages, store voterreg object in authcontext, styling fixes to voterreg flow, redirect user to appropriate page depending on their registration status and the information in the context

Devanshusp committed:
Styling + Dynamic Sidebar + Legal Pages (tos, privacy policy) (#135)

* added emailing for players and challengers

* updated emailing to include challengers

* no progress page reload if share badge earned

* signup error handling

* progress page loading animation

* making footer icons bigger

* sign in styling

* switch completion checking order in progress

* blob styling in progress page

* remove notif from header

* gives challenge badge for player taking action

* dynamic sidebar

* get challenger data in actions page without local storage

* signup error

* invite.js close icon styling

* feedback icon in header

* legal terms of servic & privacy policy

* sidebar styling

* styling changes

* voterreg errors on signup + voterreg completion redirection based on user type

* signin and signup redirection fixed

MazharulIslam-Naim committed:
Update /actions page #75 (Finish): Added in share. (#130)

* Added in Preview pop-up in the invite component. Added alt prop to images in instagram, invite, and playerwelcome compoents. Fixed a bug in playerwelcome page that wasn't geting the correct information of the person that the player is doing actions for.

* Fixed the styling that I messed up for sign in buttons on the playerwelcome page.

* Added header and footer to the preview pop-up and made sure their buttons don't work when on preview. Fixed up styling issues with preview. Added comments to footer, header, instagram, invite, and preview components.

* Fixed bug with navbar being in front of main content but invisible. In share above the player link, change text to include the challenger's name. Fixed bug where the challengerinfo was being changed on the invite component causeing the share link to be wrong.

* Added a react library to dynamicaly change the meta tags. Added in all the different images to be shared on socail media. If the user is loggedin and there is no challengerinfo in local storage then actions component uses the loggedin user refferedby userId.

* Imports for playerwelcome were not correctly tracked. Header report bugs link changed to button.

* Fixed styling issue for report button and preview header. Fixed invite error for share when user is not logged in.

rickyyx committed:
Remove FB/Messenger sharing icons && fix privacy policy link in sign-up page (#136)

Devanshusp committed:
Minor styling bug fixes (#138)

* mobile buttons + signin redirection

MazharulIslam-Naim committed:
Fixed wrong logo file name in manifest and index.html. Added alt props to imgs in header, sidebar, homepage, invite, and progress. Fixed spelling error on homepage and section heading that was incorrectly commented out. (#137)

dvorakjt committed:
Fix election reminders loop bug (#139)

* fixed election reminders bug

CaptainExtremis committed:
[PR1] Change "Go to Instagram" button, remove "8by8/" (#141)

MazharulIslam-Naim committed:
Bug fixes. (#140)

* Fixed a bug with actions page not haveing anything in localStorage. Moved challenger crown slightly over. Fixed a bug where signin would infintly loop. Made sure take action shows up for the correct users. Fixed bug with invite preview not loading.

* Moved loading end to after the challengers info is gotten. Removed success endpoint because mata tags didn't appear as intended.

arniber21 committed:
Fix Underline Issue #122 (#143)

MazharulIslam-Naim committed:
Design Fixes (#142)

* Changed line height on challengerWelcome page for the paragraphs taht are in caps. Added space between get started button and already have an account. Made the font size for registering to vote for the challenger. Made email and text icons samller. Set limit to number of characters to appear for the name of user on challenger badges and actions page. Hid the sign in messages on playerwelcome when the user is signed in. When the challenge has failed the invite friends button changes to restart challenge.

* Made sure the restart challenge button causes the invite button to appear after restart is clicked.

zjiao19 committed:
Change margin of Election Alerts page. Change link styles to bold and underline in black. (#145)

* change margin of Election Alerts page

* add margin to the bigger container

* Change link styles to bold and underline in black

* update margin for reminders-completed

Devanshusp committed:
Copy Notification (#146)

* added emailing for players and challengers

* updated emailing to include challengers

* no progress page reload if share badge earned

* signup error handling

* progress page loading animation

* making footer icons bigger

* sign in styling

* switch completion checking order in progress

* blob styling in progress page

* remove notif from header

* gives challenge badge for player taking action

* dynamic sidebar

* get challenger data in actions page without local storage

* signup error

* invite.js close icon styling

* feedback icon in header

* legal terms of servic & privacy policy

* sidebar styling

* styling changes

* voterreg errors on signup + voterreg completion redirection based on user type

* signin and signup redirection fixed

* preview

* mobile buttons + signin redirection

* copy notification

* copy notification on mobile

* copy notif text change

* link notif pointer clickthrough

MazharulIslam-Naim committed:
Made preview responsive and look more like designs. Added in challenger won invite. (#149)

wcouturiaux committed:
Tech-Design UX Walkthrough Item 9 partial (#147)

* Add react-meta-tags package

* Change signup form to html with floating labels from react-bootstrap

* Change error check to strict equal

* Update underline class text

* Bug fix on sign up if field left blank got firebase error

MazharulIslam-Naim committed:
Removed meta tags from index.html and added them individually to each endpoint. (#152)

MazharulIslam-Naim committed:
Revert "Removed meta tags from index.html and added them individually to each endpoint. (#152)" (#154)

This reverts commit 0ae281d19ec71fc1ea157a19c9ae493d20e1c524.

arniber21 committed:
Organize Folders, issue #116 (#144)

* Fix the Underline bug

* Remove .idea folder from git

* Refactor into folders

* A small bug in the styling

* Refactored more items into folders so that it's easier to tell what each component is used/used for.

* Fix some bugs with the image paths not changing

* try to fix stuff

* Update Actions.js

* Fix some git errors

projectr495836 committed:
jumping (#156)

* jumping

jump to progress page

* simplified syntax

Co-authored-by: Devanshu Sen Pandey <66650721+Devanshusp@users.noreply.github.com>

MazharulIslam-Naim committed:
Fixed a lot of the console errors and warnings. (#170)

* Fixed a lot of the console errors and warnings.

* Saw that there was copies of some components and they weren't used anywhere so removed them. Also, fixed some files code styling.

* Extra copy of chellengerwelcome.js

Co-authored-by: Devanshu Sen Pandey <66650721+Devanshusp@users.noreply.github.com>

rajnawal committed:
Built a page, added links and new info (#171)

* Built a page, added links and new info

Built the Why 8BY8 page, added required links on 3 pages and the sidebar menu. Made the required edits to the player welcome and challenger welcome pages

* spacing and padding changes in /why8by8

* Repositioned links and fixed blob rendering issue

* spacing updates

* styling /why8by8 page

* delete /why8by8.scss file

* style file name change

Co-authored-by: Devanshu <devanshusenp@gmail.com>

dvorakjt committed:
Issue #155 - Add Address Validation to Voter Reg Address Form (#173)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

* fixed a few bugs in the voter registration flow

* fixed merge conflict in package.json

* uncommented code that checks if the user has already registered to vote

* separate voter registration flow into individual pages, store voterreg object in authcontext, styling fixes to voterreg flow, redirect user to appropriate page depending on their registration status and the information in the context

* fixed election reminders bug

* add call to address validation endpoint to address block component, rename optional fields in voter registration data to correspond to newly added api fields

MazharulIslam-Naim committed:
Partner database functions (#175)

* Added CRUD functions for partner database.

* Fixed a small bug with the dates not showing up on the database if given empty strings.

arniber21 committed:
Issue 165, US4.2: Build /rewards page; Issue 167, US 6.1/8.1: Build /choosereward page (#172)

* Start work on Rewards page

* Conditionally render partners

* Add placeholder images, final styling

* Fix Authentication Bugs

* Add the ChooseReward page skeleton

* Update ChooseReward so that data is retrieved from the database

* Finalize changes

* Implement changes from PR feedback

* Fix Underline issue (from design meeting)

* Fix Font-weight issue (from design meeting)

* Fixed an issue where the site would try to render too many times while grabbing partner data.

zjiao19 committed:
Save voter registration data in Firebase (#176)

* change margin of Election Alerts page

* add margin to the bigger container

* Change link styles to bold and underline in black

* update margin for reminders-completed

* save voter reg info to firebase

* fix import file path

* update default value of VoterRegistrationData

wcouturiaux committed:
Iss158 update vote reg comp (#178)

* Add new voterreg comp elements

* Add more info container

* Add reminder and expect modals

* Implement light/dark mode for PopupModal

* Update reminder modal style

* Add Reg by mail modal and style what to expect modal

* Update styling for What to expect modal

* Update styling and add links

* get saved data from database

Co-authored-by: Devanshu <devanshusenp@gmail.com>

dvorakjt committed:
Update /actions and address error handling (#179)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

* fixed a few bugs in the voter registration flow

* fixed merge conflict in package.json

* uncommented code that checks if the user has already registered to vote

* separate voter registration flow into individual pages, store voterreg object in authcontext, styling fixes to voterreg flow, redirect user to appropriate page depending on their registration status and the information in the context

* fixed election reminders bug

* add call to address validation endpoint to address block component, rename optional fields in voter registration data to correspond to newly added api fields

* updated actions page to add links upon registering to vote, added general use-case error modal, updated design of address error messsages, cleaned up a few things

rajnawal committed:
Worked on Issue 8.1 (#182)

* Worked on issue 8.1

Added the required functionality to the /choosereward, /progress, and /actions pages.

* bug style fixes

Co-authored-by: Devanshu Pandey <devanshusenp@gmail.com>

Devanshusp committed:
Email verification (#181)

* redid signup page

* email verification after signup + signup form cleanup

* private route verification success page

* signin verification email

* signin verificaiton email logic

* /verify page routing

* firebase functions

dvorakjt committed:
Revert Actions, Progress and Welcome pages to Alpha if no partners exist, fix visual bugs (#184)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

* fixed a few bugs in the voter registration flow

* fixed merge conflict in package.json

* uncommented code that checks if the user has already registered to vote

* separate voter registration flow into individual pages, store voterreg object in authcontext, styling fixes to voterreg flow, redirect user to appropriate page depending on their registration status and the information in the context

* fixed election reminders bug

* add call to address validation endpoint to address block component, rename optional fields in voter registration data to correspond to newly added api fields

* updated actions page to add links upon registering to vote, added general use-case error modal, updated design of address error messsages, cleaned up a few things

* added partners context, fixed many visual bugs

ty730 committed:
Update challenge won email for issue #168 (#186)

* IG and SMS Invite/Share workflow, Issue #70

Created IG, SMS flow for Invite/Share, Issue #70

* Made link on one line, overflow hidden, Issue #70

* Fixed header text turning black

* Update challenger won email

* isChallenger for challengeWon email

dvorakjt committed:
Update Rewards Page with Current Partners (#188)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

* fixed a few bugs in the voter registration flow

* fixed merge conflict in package.json

* uncommented code that checks if the user has already registered to vote

* separate voter registration flow into individual pages, store voterreg object in authcontext, styling fixes to voterreg flow, redirect user to appropriate page depending on their registration status and the information in the context

* fixed election reminders bug

* add call to address validation endpoint to address block component, rename optional fields in voter registration data to correspond to newly added api fields

* updated actions page to add links upon registering to vote, added general use-case error modal, updated design of address error messsages, cleaned up a few things

* added partners context, fixed many visual bugs

* updated rewards page

dvorakjt committed:
Fix Potential Security Issue with Voter Registration Data (#189)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

* fixed a few bugs in the voter registration flow

* fixed merge conflict in package.json

* uncommented code that checks if the user has already registered to vote

* separate voter registration flow into individual pages, store voterreg object in authcontext, styling fixes to voterreg flow, redirect user to appropriate page depending on their registration status and the information in the context

* fixed election reminders bug

* add call to address validation endpoint to address block component, rename optional fields in voter registration data to correspond to newly added api fields

* updated actions page to add links upon registering to vote, added general use-case error modal, updated design of address error messsages, cleaned up a few things

* added partners context, fixed many visual bugs

* updated rewards page

* user voter registration data is now saved to and looked up from a separate collection in the database.

ty730 committed:
Send welcome email after email verification (#191)

* IG and SMS Invite/Share workflow, Issue #70

Created IG, SMS flow for Invite/Share, Issue #70

* Made link on one line, overflow hidden, Issue #70

* Fixed header text turning black

* Update challenger won email

* isChallenger for challengeWon email

* Move challengerWelcome email to after email verify

rajnawal committed:
Made changes to the choose rewards flow (#193)

* Worked on issue 8.1

Added the required functionality to the /choosereward, /progress, and /actions pages.

* bug style fixes

* Styling changes and correctly displaying data on the coupons

Co-authored-by: Devanshu Pandey <devanshusenp@gmail.com>

dvorakjt committed:
Promisify Firebase Https Functions, link Voter Reg FormCompleted page to /actions page (#190)

* integrate RTV Election Reminder form

* Added voter registration route

* styled election reminders form, made updates to voterregistrationform

* election reminder form now updates user document and checks if the user has already signed up for reminders, also redirects to actions

* voter registration form visual flow complete

* integrated voter reg frontend with api, fixed bug in frontend

* made some small improvements to the UI

* voter registration flow now includes tooltips and registration status check

* voter registration flow ready for end-to-end testing

* updated styling in election reminders and voter registration, loading wheel can now be set to a block-style display

* eligibility page now re-evaluates elibility if dob is updated after the zip has been entered and elibility has already been determined

* resolved conflict in package-lock.json

* updated election reminders loading screen and loading wheel component

* fixed a few bugs in the voter registration flow

* fixed merge conflict in package.json

* uncommented code that checks if the user has already registered to vote

* separate voter registration flow into individual pages, store voterreg object in authcontext, styling fixes to voterreg flow, redirect user to appropriate page depending on their registration status and the information in the context

* fixed election reminders bug

* add call to address validation endpoint to address block component, rename optional fields in voter registration data to correspond to newly added api fields

* updated actions page to add links upon registering to vote, added general use-case error modal, updated design of address error messsages, cleaned up a few things

* added partners context, fixed many visual bugs

* updated rewards page

* user voter registration data is now saved to and looked up from a separate collection in the database.

* Promisify firebase https functions to aid in debugging

* update go to challenge link after completing voter reg to link to actions page

Devanshusp reviewed (changes requested):
The emailing works as specified, just a few merge conflicts, will merge once fixed

Thanks @dvorakjt!!

MazharulIslam-Naim committed:
Added Google analytics. (#197)

MazharulIslam-Naim committed:
Fix to a bunch of ESLint warnings. (#200)

MazharulIslam-Naim committed:
Fixed more console warnings. (#202)

Removing unused imports/variables, added/fixing img tag alt text, addressing code style issues like use of != over !==, use of class over className, changed links to buttons and fixed useEffect arrays.
